src/verifier/step4/SumcheckLogic.sol

```diff
@@ -0,0 +1,488 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "src/pasta/Pallas.sol";
+import "src/pasta/Vesta.sol";
+import "src/verifier/step4/KeccakTranscript.sol";
+import "src/verifier/step4/EqPolynomial.sol";
+
+library PallasPolyLib {
+    struct MLPoly {
+        uint64 numvars;
+        uint256[] Z;
+    }
+
+    struct SparseEntry {
+        uint256 idx;
+        uint256 entry;
+    }
+
+    struct SparsePoly {
+        uint64 numvars;
+        SparseEntry[] Z;
+    }
+
+    struct UniPoly {
+        uint256[] coeffs;
+    }
+
+    struct CompressedUniPoly {
+        uint256[] coeffs_except_linear_term;
+    }
+
+    function newML(uint256[] memory Z, uint64 numvars) public pure returns (MLPoly memory) {
```

Comment:
I think we don't need two identical `newML` functions in PallasPolyLib and VestaPolyLib as they don't need to know the curve